crypto/src/contribution.rs

```diff
@@ -19,8 +20,12 @@ impl Contribution {
     }
 
     /// Adds entropy to this contribution. Can be called multiple times.
+    /// The entropy is consumed and the blob is zeroized after use.
     #[instrument(level = "info", skip_all, , fields(n1=self.powers.g1.len(), n2=self.powers.g2.len()))]
-    pub fn add_entropy<E: Engine>(&mut self, entropy: [u8; 32]) -> Result<(), CeremonyError> {
+    pub fn add_entropy<E: Engine>(&mut self, entropy: &mut [u8; 32]) -> Result<(), CeremonyError> {
```

Comment:
Having this as a mutable reference feels strange. If it needs to be burned I would expect the function to receive full ownership (so pass by value).

Reply:
I agree. One reason this is done in some codebases is to attempt to stop the compiler from doing a memcpy of the array. However, it's not guaranteed as the compiler can do whatever it pleases